File: tools/bench/polyglot/javascript/spatial_grid_bvh.js
function intersect(a, b) {
    if (a.maxX < b.minX || a.minX > b.maxX) return false;
    if (a.maxY < b.minY || a.minY > b.maxY) return false;
    return true;
}

const target = { minX: 10.0, minY: 10.0, maxX: 20.0, maxY: 20.0 };
let hits = 0;
for (let i = 0; i < 1000000; i++) {
    const probe = { minX: 15.0, minY: 15.0, maxX: 25.0, maxY: 25.0 };
    if (intersect(target, probe)) hits++;
}
console.log(hits.toString());

File: tools/bench/polyglot/c/spatial_grid_bvh.c
#include <stdalign.h>
#include <stdbool.h>
#include <stdio.h>

typedef struct {
    double min_x, min_y, max_x, max_y;
} __attribute__((aligned(16))) AABB;

static bool intersect(AABB a, AABB b) {
    if (a.max_x < b.min_x || a.min_x > b.max_x) return false;
    if (a.max_y < b.min_y || a.min_y > b.max_y) return false;
    return true;
}

int main(void) {
    AABB target = {10.0, 10.0, 20.0, 20.0};
    int hits = 0;
    for (int i = 0; i < 1000000; i++) {
        AABB probe = {15.0, 15.0, 25.0, 25.0};
        if (intersect(target, probe)) hits++;
    }
    printf("%d\n", hits);
    return 0;
}
